### PR TITLE
(1CT) Create One Click Trading Session

### PR DESCRIPTION
**Pull request by JoseRFelix**
### ClickUp Task

[1CT Logic](https://app.clickup.com/t/86a21b84u)
[1CT Profile Link](https://app.clickup.com/t/86a21b7z1)

## Brief Changelog

- Allow Creating 1CT session
- Remove Session from LocalStorage after disconnect
- Add 1CT Expiry and Activated toasts
- Add 1CT Intro Modal
- Display the 'welcome back screen' in the wallet modal after the user has started at least one 1CT session
- Display the remaining session time radial progress in the navbar

## Testing and Verifying

- [ ] Should allow user to create 1CT account
- [ ] Should allow to create a 1CT session with different parameters
- [ ] Should display a toast when the session expires
- [ ] Should display a toast when the session starts

### CODE REVIEW

File: packages/stores/src/account/osmosis/index.ts [hunk 1]
```diff
@@ -2125,21 +2125,77 @@ export class OsmosisAccountImpl {
...
+            .balances.forEach((balance) => balance.waitFreshResponse());
+
+          queries.cosmos.queryDelegations
+            .getQueryBech32Address(this.address)
+            .waitFreshResponse();
+
+          queries.cosmos.queryRewards
+            .getQueryBech32Address(this.address)
+            .waitFreshResponse();
+        }
```
Comment (jonator) [inline packages/stores/src/account/osmosis/index.ts:2177]: I'm assuming we'll eventually remove these for when we finish tRPC migration?

File: packages/types/src/autheticator-types.ts
Comment (jonator) [inline packages/types/src/autheticator-types.ts:1]: TBH I'm not a fan of this types package, similar to not being a fan of types.ts files. I think types are easier to understand when they're defined closer to where they are used. Thoughts?

File: packages/web/components/buttons/button.tsx
Comment (jonator) [inline packages/web/components/buttons/button.tsx:1]: I think this button will be deprecated in favor of the new ui/button cc @mattupham

**Review by jonator — COMMENTED** [review]
I think build is failing, but I added an initial review with high-level thoughts

File: packages/stores/src/account/osmosis/index.ts (hunk 1 above)
Comment (JoseRFelix) [reply packages/stores/src/account/osmosis/index.ts:2177]: That's right. In the future, I plan to eliminate these injected accounts and create message-composing functions. This way, we'll only need one function for broadcasting. This approach will simplify the process and offer the greatest flexibility for creating transactions, whether they have a single or multiple messages.

File: packages/types/src/autheticator-types.ts
Comment (JoseRFelix) [reply packages/types/src/autheticator-types.ts:1]: I agree, however, we need a way to share the types between the `stores` and `web` packages. I think we can remove this package once we have the `server` package where we can co-locate these types with their utils.

File: packages/web/components/buttons/button.tsx
Comment (JoseRFelix) [reply packages/web/components/buttons/button.tsx:1]: Good to know! I'll update this

**Review by DavideSegullo — COMMENTED** [review]
I'm trying to initialized a session, but I get an error here:
getAccountPubKeyAndAuthenticators

Screenshot 2024-03-01 alle 11 05 13

**Comment by JoseRFelix** [comment]
> I'm trying to initialized a session, but I get an error here: getAccountPubKeyAndAuthenticators
> 
> Screenshot 2024-03-01 alle 11 05 13

Previews will not work until we get an HTTPS link for the smart accounts node. I'll make sure to keep you posted once I have it. In the meantime, I would appreciate a code review on this PR

File: packages/web/components/alert/toast.tsx [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
-import { FunctionComponent } from "react";
+import React, { FunctionComponent } from "react";
 import { toast, ToastOptions } from "react-toastify";
 
 import { Alert, ToastType } from "~/components/alert";
 import { Icon } from "~/components/assets";
 import { t } from "~/hooks";
 
+export function dismissAllToasts() {
+  toast.dismiss();
```
Comment (DavideSegullo) [inline packages/web/components/alert/toast.tsx:10]: Why do we need a different function for that? Do you assume you will need to add more shared logic here in the future?

File: packages/web/components/one-click-trading/screens/reset-period-screen.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  "day",
+  "week",
+  "month",
+  "year",
+] as OneClickTradingResetPeriods[];
+
+export function getResetPeriodTranslationKey(
+  id: OneClickTradingResetPeriods
+): string {
+  switch (id) {
```
Comment (DavideSegullo) [inline packages/web/components/one-click-trading/screens/reset-period-screen.tsx:19]: If the key is always in the same format as the corresponding translation, couldn't we just check if "ResetPeriods" contains that key and then construct the string as "oneClickTrading.settings.resetPeriodScreen.periods.{key}"? Obviously if it doesn't at contains we return "oneClickTrading.settings.resetPeriodScreen.periods.unknown", so in the future if we add other keys we only need to update "ResetPeriods" and the translations and we won't have a long switch.

What do you think?

File: packages/web/components/one-click-trading/screens/session-period-screen.tsx [hunk 4]
```diff
@@ -1,40 +1,45 @@
...
-    periodTranslationKey:
-      "oneClickTrading.settings.sessionPeriodScreen.periods.month",
-  },
-  {
-    id: "year",
-    periodTranslationKey:
-      "oneClickTrading.settings.sessionPeriodScreen.periods.year",
-  },
-] as const;
+export function getSessionPeriodTranslationKey(
```
Comment (DavideSegullo) [inline packages/web/components/one-click-trading/screens/session-period-screen.tsx:9]: How about we do the same thing proposed above with "ResetPeriods" here as well?

**Review by DavideSegullo — APPROVED** [review]

File: packages/web/components/one-click-trading/screens/reset-period-screen.tsx (hunk 3 above)
Comment (JoseRFelix) [reply packages/web/components/one-click-trading/screens/reset-period-screen.tsx:19]: This is a good idea! I'll add this in

**Pull request merged**